tests/ui/resolve/open-ns-3.rs

```diff
@@ -1,8 +1,10 @@
 //@ aux-crate: my_api=open-ns-mod-my_api.rs
 //@ aux-crate: my_api::utils=open-ns-my_api_utils.rs
+//@ compile-flags: -Z namespaced-crates
+//@ edition: 2024
+//@ build-pass
```

Comment:
This should fail with `error[E0260]: the name `utils` is defined multiple times`, because `open-ns-mod-my_api.rs` also has a module `utils` which conflicts with the namespaced crate `my_api::utils`. Sorry for not having included the error annotation, but I got distracted when I hit the problem with the compiler tests.

Reply:
Ah, no worries. I'll remove my changes here and see if we're getting a reasonable error message.

I think a similar test to my changes here would still be useful, especially since it seems like looking up a path in expression context behaves differently from in a `use` statement.

Reply:
Okay, I just pushed a new change to open-ns-3 that I think captures the intent better.